### DISCUSSION TITLE
I want to contribute something that is not that big of a deal

### DISCUSSION THREAD

**Discussion by HorrorBest** [electronstudio/raylib-python-cffi#41] [Ideas]
Well I was looking around the binding and The colors are few to say the least, And It doesn't support **"Hex code"** So I thought It might help If I add a function that gets a Hex code as a color also I would Like to add the 140 default colors that come with the **HTML language** So people have a lot more colors to work with.

Edit: The hex code could have Transparency Info and It would still work the way it is intended to work*

**Comment by electronstudio** [comment]
Raylib does support hex code colours with the GetColor function, but Raylib expects them to be unsigned while Python ints are signed so you have to convert.  Might be worth modifying Pyray to do this automatically.  For now here is a demo of how to do it

```
from raylib.pyray import PyRay
pyray = PyRay()
pyray.init_window(800, 400, 'demo')

def to_unsigned(i):
    if i < 2**31:
        return i
    else:
        return i - 2**32

white = pyray.get_color(to_unsigned(0xFFFFFFFF))
red = pyray.get_color(to_unsigned(0xFF0000FF))
green = pyray.get_color(to_unsigned(0x00FF00FF))
blue = pyray.get_color(to_unsigned(0x0000FFFF))

while not pyray.window_should_close():
    pyray.begin_drawing()
    pyray.clear_background(white)
    pyray.draw_rectangle(0, 0, 100, 100, red)
    pyray.draw_rectangle(100, 100, 100, 100, green)
    pyray.draw_rectangle(200, 200, 100, 100, blue)
    pyray.end_drawing()
pyray.close_window()

```

**Reply by HorrorBest** [reply]
I put a function in the Colors.py file and here it is and It works You can use it to convert hex to RGBA

> def Hex(code):
    x = code.lstrip('#')
    if len(code) == 8:
        RGBA = tuple(int(x[i:i + 2], 16) for i in (0, 2, 4, 6))
    else:
        h += 'FF'
        RGBA = tuple(int(x[i:i+2], 16) for i in (0, 2, 4, 6))
    return RGBA

and to use it you just do (with or without the #):

> pyray.draw_rectangle(200, 200, 100, 100, hex('#0DG254'))

**Reply by electronstudio** [reply]
That's great, but as I posted in my example, Raylib already has a function that does that, `get_color()`

The only issue with `get_color()` is the need for `to_unsigned()`, so I have contacted Raylib about that.... https://github.com/raysan5/raylib/pull/1954

That means once Raylib 4.0 is released we won't need `to_unsigned()` anymore.

**Reply by electronstudio** [reply]
in fact it might work in the next version i release very soon, even with 3.7.

**Reply by electronstudio** [reply]
Should be fixed now in current version (update with pip.)  So this should now work:

```
from raylib import PyRay
pyray = PyRay()
pyray.init_window(800, 400, 'demo')


white = pyray.get_color(0xFFFFFFFF)
red = pyray.get_color(0xFF0000FF)
green = pyray.get_color(0x00FF00FF)
blue = pyray.get_color(0x0000FFFF)

while not pyray.window_should_close():
    pyray.begin_drawing()
    pyray.clear_background(white)
    pyray.draw_rectangle(0, 0, 100, 100, red)
    pyray.draw_rectangle(100, 100, 100, 100, green)
    pyray.draw_rectangle(200, 200, 100, 100, blue)
    pyray.end_drawing()
pyray.close_window()
```

**Reply by HorrorBest** [reply]
thats not a HEX code well it does have the transparency can you edit it so it automatically puts max transparency if there is no Transparency value in the Hex code, I could do that But Idk where it is in the project

**Comment by electronstudio** [comment]
As for adding pre-defined colours, we use the same list that Raylib does.  If you think there should be more then you would need to get them added at https://github.com/raysan5/raylib/blob/7749a2805c0f28762c206da0b7c954cbe9e318e2/src/raylib.h#L145

You can  publish your own separate module with your own new colours in it of course.

**Reply by HorrorBest** [reply]
well I asked in discord And they said its fine to add new colors as long as you ( the one who made the bind ) agree

**Reply by electronstudio** [reply]
The purpose of this binding is to be (as close as possible to) 100% compatible with Raylib, so I’m not going to agree to any changes or extra features that make it deviate from what’s in Raylib.

**Reply by HorrorBest** [reply]
not even the Hex color support function and It would better be put with the "colors.py" cause its a color related function ??
